src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+                uint64 magnitudeToDeallocate = uint64(currentMagnitude) - allocation.magnitudes[i];
+                _queuedDeallocationIndices[operator][allocation.strategy][operatorSetKey].push(
+                    _pendingFreeMagnitude[operator][allocation.strategy].length
+                );
+                _pendingFreeMagnitude[operator][allocation.strategy].push(
+                    PendingFreeMagnitude({
+                        magnitudeDiff: magnitudeToDeallocate,
+                        completableTimestamp: deallocationCompletableTimestamp
+                    })
+                );
+
+                // 2. decrement allocated magnitude
+                magnitudeUpdates.decrementAtAndFutureSnapshots({
+                    key: uint32(block.timestamp),
+                    decrementValue: magnitudeToDeallocate
+                });
+            } else if (allocation.magnitudes[i] > uint64(currentMagnitude)) {
+                // Newly configured magnitude is greater than current value.
+                // Therefore we handle this as an allocation
+
+                // 1. decrement free magnitude by incremented amount
+                uint64 magnitudeToAllocate = allocation.magnitudes[i] - uint64(currentMagnitude);
+                OperatorMagnitudeInfo storage info = operatorMagnitudeInfo[operator][allocation.strategy];
+                require(info.freeMagnitude >= magnitudeToAllocate, InsufficientAllocatableMagnitude());
+                info.freeMagnitude -= magnitudeToAllocate;
+
+                // 2. allocate magnitude which will take effect in the future 21 days from now
+                magnitudeUpdates.push({key: allocationEffectTimestamp, value: allocation.magnitudes[i]});
+            }
+        }
+    }
+
+    /**
+     * @notice Get the number of queued dealloations for the given (operator, strategy, operatorSetKey) tuple
+     * @param operator address to get queued deallocations for
+     * @param strategy the strategy to get queued deallocations for
+     * @param operatorSetKey the encoded operatorSet to get queued deallocations for
+     */
+    function _getNumQueuedDeallocations(
+        address operator,
+        IStrategy strategy,
+        bytes32 operatorSetKey
+    ) internal view returns (uint256) {
+        uint256 numQueuedDeallocations;
+
+        uint256 length = _queuedDeallocationIndices[operator][strategy][operatorSetKey].length;
+        for (uint256 i = length; i > 0; --i) {
+            // index of pendingFreeMagnitude/deallocation to check for slashing
+            uint256 index = _queuedDeallocationIndices[operator][strategy][operatorSetKey][i - 1];
+
+            // If completableTimestamp is greater than completeUntilTimestamp, break
+            if (block.timestamp < _pendingFreeMagnitude[operator][strategy][index].completableTimestamp) {
+                ++numQueuedDeallocations;
+            } else {
+                break;
+            }
```

Comment:
nit: comment refers to `completeUntilTimestamp` which doesn't exist (any more) -- seems like this is now just the current time?